Detectors/TOF/reconstruction/src/Encoder.cxx

```diff
@@ -31,99 +31,67 @@ Encoder::Encoder()
 {
   // intialize 72 buffers (one per crate)
   for (int i = 0; i < 72; i++) {
-    mIntegratedBytes[i] = 0;
     mBuffer[i] = nullptr;
     mUnion[i] = nullptr;
     mStart[i] = nullptr;
-    mNRDH[i] = 0;
+    mCrateOn[i] = true;
   }
 }
 
 void Encoder::nextWord(int icrate)
 {
   if (mNextWordStatus[icrate]) {
-    nextWordNoEmpty(icrate);
+    mUnion[icrate]++;
     mUnion[icrate]->data = 0;
-    nextWordNoEmpty(icrate);
+    mUnion[icrate]++;
     mUnion[icrate]->data = 0;
   }
-  nextWordNoEmpty(icrate);
-  mNextWordStatus[icrate] = !mNextWordStatus[icrate];
-}
-
-void Encoder::nextWordNoEmpty(int icrate)
-{
   mUnion[icrate]++;
-
-  // check if you went over the buffer size
-  int csize = getSize(mRDH[icrate], mUnion[icrate]);
-  if (csize >= Geo::RAW_PAGE_MAX_SIZE) {
-    addPage(icrate);
-  }
+  //nextWordNoEmpty(icrate);
+  mNextWordStatus[icrate] = !mNextWordStatus[icrate];
 }
 
 bool Encoder::open(const std::string name, std::string path)
 {
   bool status = false;
 
-  for (int i = 0; i < NCRU; i++) {
-    std::string nametmp;
-    nametmp.append(Form("%s/cru%02d", path.c_str(), i));
-    nametmp.append(name);
-    printf("TOF Raw encoder: create stream to CRU: %s\n", nametmp.c_str());
-    if (mFileCRU[i].is_open()) {
-      std::cout << "Warning: a file (" << i << ") was already open, closing" << std::endl;
-      mFileCRU[i].close();
-    }
-    mFileCRU[i].open(nametmp.c_str(), std::fstream::out | std::fstream::binary);
-    if (!mFileCRU[i].is_open()) {
-      std::cerr << "Cannot open " << nametmp << std::endl;
-      status = true;
-    }
+  // register links
+  o2::header::RAWDataHeader rdh;
+  for (int feeid = 0; feeid < 72; feeid++) {
+    // cru=0 --> FLP 0, endpoint 0 --> 18 links -> fees 0-17
+    // cru=1 --> FLP 0, endpoint 1 --> 18 links -> fees 18-35
+    // cru=2 --> FLP 1, endpoint 0 --> 18 links -> fees 36-53
+    // cru=3 --> FLP 1, endpoint 1 --> 18 links -> fees 54-71
+    rdh.feeId = feeid;
+    rdh.cruID = feeid / NLINKSPERCRU;
+    rdh.linkID = feeid % NLINKSPERCRU;
+    rdh.endPointID = rdh.cruID % 2;
+    // currently storing each CRU in a separate file
+    if (mCrateOn[feeid])
```

Comment:
Pietro told me that you are also setting ``RDH.detectorField``? Is it constant data? You can add it to RDH here, then it will be propagated further.

Reply:
I will check with Pietro. I don't know about that.
In the meanwhile I would add 
writer.setAddSeparateHBFStopPage(false)
Thanks for the suggestion

Reply:
It seems that 
setAddSeparateHBFStopPage(false)
has some problems when raw are digested by tof-compressor.
Therefore, I keep the default value.
From my side we can merge since, in any case, detectorField is not mandatory.

Reply:
OK, apparently tof-compressor expects separate page, merging.